src/lib.rs

```diff
@@ -173,6 +173,26 @@ impl<T> Cactus<T> {
     pub fn vals(&self) -> CactusValsIter<T> {
         CactusValsIter{next: self.node.as_ref()}
     }
+
+    /// Consume this Cactus node and return its data. If the cactus node has no children, its data
+    /// is returned without cloning; if the node has children then the internal data is cloned
+    /// (hence why `T` must implement the `Clone` trait).
+    ///
+    /// # Examples
+    /// ```
+    /// use cactus::Cactus;
+    /// let c = Cactus::new().child(1).child(2);
+    /// let p = c.parent().unwrap();
+    /// assert_eq!(c.take_or_clone_val(), Some(2));
+    /// // At this point c has been consumed and can no longer be referenced.
+    /// assert_eq!(p.val(), Some(&1));
+    /// ```
+    pub fn take_or_clone_val(self) -> Option<T> where T: Clone {
```

Comment:
Does this mean we can only store things that implement the `Clone` trait? Is that OK?